src/bucket/InMemoryIndex.cpp

```diff
@@ -59,8 +60,9 @@ InMemoryIndex::InMemoryIndex(BucketManager const& bm,
     std::streamoff lastOffset = 0;
     std::optional<std::streamoff> firstOffer;
     std::optional<std::streamoff> lastOffer;
+    SHA256* hasherPtr = hasher.has_value() ? &hasher.value() : nullptr;
```

Comment:
nit: Why not pass the pointer all around? Pointer is still the best we have now for `optional<reference>` intended here.